src/main/scala/scalismo/faces/io/ply/PLYMesh.scala

```diff
@@ -140,10 +140,16 @@ object PLYMesh {
     val triangleMesh3D = TriangleMesh3D(points, triangles)
 
     val colors = getSurfaceColor(vertexProperties, faceProperties, texture, triangles)
-    val normals = getSurfaceNormal(vertexProperties, faceProperties, texture, triangles)
 
-    // building the mesh class
-    ColorNormalMesh3D(triangleMesh3D, colors, normals)
+    Try { // use stored vertex normals if available
+      val normals = getSurfaceNormal(vertexProperties, faceProperties, texture, triangles)
+      ColorNormalMesh3D(triangleMesh3D, colors, normals)
+
+    }.getOrElse { // compute the vertex normals using the built mesh - note: this will fail if vertices without adjacent triangles exist
+      val normals = triangleMesh3D.vertexNormals
```

Comment:
For cases that do have "floating" vertices, it might be useful to add a flag to the read method saying "read anyway" .. because in the current case, it will not be possible to read such a mesh until one fixes it with an external tool (removing non-manifold vertices).

Reply:
Internally we rely often on proper meshes. So I think it is not a good idea to offer convenient methods to read meshes which after loading you can for example not render (at least not using vertex normals). In addition, we do not have yet a good set of tools to repair meshes. Which, in my opinion, is also a reason why we should prevent the user to start of working with corrupted meshes.

So when you are really willing to circumvent the safety mechanisms of the conveniece methods for reading a mesh, then you can read the data from a ply file using a snippet like the one that follows and build your mesh out of the parts you have read. In this way you can correct some values before creating the mesh and hence repair it during loading before you use it.

```scala
  val (plyValues, textures) = PlyReader.read("test.ply")
  val (vertexProperties, faceProperties) = PlyMeshReader.getProperties(plyValues)
  val vertices = PlyMeshReader.getVertices(vertexProperties)
  val triangles = PlyMeshReader.getTriangles(faceProperties)
  val triangleMesh3D = TriangleMesh3D(vertices, triangles)
  val colors = PlyMeshReader.getSurfaceColor(vertexProperties, faceProperties, textures, triangles)
```

Reply:
Ok. Other than that am fine to merge.